Restore testing of jbmc

smowton opened the pull request:
This was accidentally disabled when rearranging the repository due to the jbmc directory being included before enable_testing() was called.

smowton committed:
Restore testing of jbmc

tautschnig commented:
Tests are failing with "failed to figure out type of file", and I find it surprising that that's only happening in the CMake world.

peterschrammel commented:
It's calling the wrong executable, it seems.

peterschrammel commented:
https://github.com/diffblue/cbmc/blob/develop/jbmc/regression/jdiff/CMakeLists.txt#L2: goto-diff ->  jdiff
https://github.com/diffblue/cbmc/blob/develop/jbmc/regression/janalyzer-taint/CMakeLists.txt#L2: goto-analyzer -> janalyzer

smowton commented:
Updated accordingly

smowton commented:
Now failing due to concurrency of the jbmc and jbmc-symex-driven-loading profiles, which had always been in conflict but never happened to run concurrently before.

smowton committed:
JBMC tests: suffix logfiles when using symex-driven loading

The three test directories jbmc, jbmc-strings and strings-smoke-tests all have a
second test profile that uses --symex-driven-lazy-loading, but this meant after the
test completed the logfiles from the first profile would have been overwritten, and
if very unlucky the two profiles could run concurrently and fight over the same logfile.

Therefore we now allow test.pl to be given a log-file suffix, both for its main log
and test output logs, use that facility for these tests.

smowton commented:
Pushed a fix where the log files are given profile-specific suffixes.

chrisr-diffblue reviewed (approved):
I think these changes look sane, but I haven't actually confirmed all the expected tests are now run.

peterschrammel reviewed (approved):
LGTM modulo CI passing

peterschrammel commented on jbmc/regression/jbmc-cover/Makefile:
```diff
@@ -4,9 +4,11 @@ include ../../src/config.inc
 
 test:
 	@../$(CPROVER_DIR)/regression/test.pl -p -c ../../../src/jbmc/jbmc
+        @../$(CPROVER_DIR)/regression/test.pl -p -c "../../../src/jbmc/jbmc --symex-driven-lazy-loading" -X symex-driven-lazy-loading-expected-failure -s symex-driven-loading
```
@smowton, it seems you have replaced \t by spaces here and below...

smowton replied:
Yup. Fixed.

smowton commented:
This time it's failing because I think the author of jbmc-cover never intended to make a symex-driven-loading test pass, because they used a duplicate test ID of the normal cbmc-java (now jbmc) directory (I guess they just copied the CMakeLists.txt file, but the Makefile didn't try to run symex-driven tests). However now jbmc is renamed the test activates, and fails. I've therefore disabled testing the jbmc-cover directory with symex-driven loading for now, to investigate later.
